lib/ReactViews/Preview/MappablePreview.jsx

```diff
@@ -43,25 +42,22 @@ class MappablePreview extends React.Component {
   };
 
   @action.bound
-  toggleOnMap(event) {
+  async toggleOnMap(event) {
     if (defined(this.props.viewState.storyShown)) {
       this.props.viewState.storyShown = false;
     }
 
     const keepCatalogOpen = event.shiftKey || event.ctrlKey;
     const toAdd = !this.props.terria.workbench.contains(this.props.previewed);
 
-    if (toAdd) {
-      this.props.terria.timelineStack.addToTop(this.props.previewed);
-    } else {
-      this.props.terria.timelineStack.remove(this.props.previewed);
-    }
-
-    const addPromise = addToWorkbench(
-      this.props.terria.workbench,
-      this.props.previewed,
-      toAdd
-    ).then(() => {
+    try {
+      if (toAdd) {
+        this.props.terria.timelineStack.addToTop(this.props.previewed);
+        await this.props.terria.workbench.add(this.props.previewed);
+      } else {
+        this.props.terria.timelineStack.remove(this.props.previewed);
+        this.props.terria.workbench.remove(this.props.previewed);
+      }
```

Comment:
This is common to a lot of files. Maybe it should be a function in a separate file `addToOrRemoveFromWorkbenchAndTimelineStack`.

Reply:
😑